lib/index.coffee

```diff
@@ -15,35 +15,38 @@ create_redis_client = _.memoize (host, port, password, log) ->
 
   # Client emits an error every time it tries to reconnect and fails. Only emit an error once.
   _redis.once 'error', (err) ->
-    log "RESERVE: Error connecting to REDIS: #{host}:#{port}.", err
+    logger.errorD "redis-error", { host, port, err }
     throw err
   _redis
 
 , (host, port, password) -> "#{host}:#{port}:#{password}"
 
 module.exports = class ReserveResource
-  constructor: (@by, @host, @port, @heartbeat_interval, @lock_ttl, @log=console.error, @password='') ->
+  constructor: (@by, @host, @port, @heartbeat_interval, @lock_ttl, @password='') ->
     @_lost_reservation = false
+    @identifer = process.env.JOB_ID or process.pid
+    @logger = new kayvee.logger(@by)
+    @logger.globals.job_id = process.env.JOB_ID
+    @logger.globals.via = "node-redis-reservation"
     return
 
-  lock: (resource, cb) ->
+  lock: (system_id, cb) ->
     @_init_redis()
 
-    reserve_key = "reservation-#{resource}"
-    val = "#{os.hostname()}-#{@by}-#{process.pid}"
+    reserve_key = "reservation-#{system_id}"
+    val = "#{@by}-#{@identifer}"
 
     async.waterfall [
       (cb_wf) => @_redis.get reserve_key, (err, val) =>  # log existing value for runtime debugging
-        @log "RESERVE: Existing resource lock value for", reserve_key, val unless err?  # ignore errors
+        @logger.infoD "reservation-existing", { key: reserve_key, system_id } unless err?
         cb_wf()
       (cb_wf) => @_redis.set [reserve_key, val, 'EX', @lock_ttl, 'NX'], (err, state) =>
-        debug "RESERVE: (err, state, reserve_key):", err, state, reserve_key
         if err?
-          @log "RESERVE: Failed to get lock for #{reserve_key} on #{val}"
+          @logger.infoD "reservation-failed", { system_id, key: reserve_key }
```

Comment:
Probably useful to log the error itself too (unless it's guaranteed to be logged when propagated up)